akka-samples/akka-sample-cluster-scala/src/main/scala/sample/cluster/factorial/FactorialFrontend.scala

```diff
@@ -61,10 +61,18 @@ object FactorialFrontend {
     //#registerOnRemoved
     Cluster(system).registerOnMemberRemoved {
       // exit JVM when ActorSystem has been terminated
-      system.registerOnTermination(System.exit(-1))
+      system.registerOnTermination(System.exit(0))
       // in case ActorSystem shutdown takes longer than 10 seconds,
-      // exit the JVM forcefully anyway
-      system.scheduler.scheduleOnce(10.seconds)(System.exit(-1))(system.dispatcher)
+      // exit the JVM forcefully anyway.
+      // Use a Deadline since the scheduled task will also be run when
+      // the ActorSystem is terminated normally.
+      val shutdownDeadline = Deadline.now + 10.seconds
+      system.scheduler.scheduleOnce(shutdownDeadline.timeLeft + 10.millis) {
+        if (shutdownDeadline.isOverdue())
+          System.exit(-1)
+        // otherwise it was a successful shutdown and we exit the JVM via the
+        // system.registerOnTermination hook
+      }(system.dispatcher)
```

Comment:
This is in fact very complicated. I can't find an easier way that is correct so please speak up if you have any idea.

I don't want to use Await to block inside the registerOnMemberRemoved since that would block the actor system shutdown.

Using the system termination Future with `after` doesn't help either, as far as I can see.